src/cmap/conn/mod.rs

```diff
@@ -206,11 +208,17 @@ impl Connection {
         command: Command,
         request_id: impl Into<Option<i32>>,
     ) -> Result<CommandResponse> {
+        self.commmand_executing = true;
+
         let message = Message::with_command(command, request_id.into());
         message.write_to(&mut self.stream).await?;
 
         let response_message = Message::read_from(&mut self.stream).await?;
-        CommandResponse::new(self.address.clone(), response_message)
+        let command_response = CommandResponse::new(self.address.clone(), response_message);
```

Comment:
do we need to split it up like this? I think we should be fine just setting `command_executing` to false right after the reading is done and just keep returning the `CommandResponse` construction.

Reply:
I've moved the assignments of `self.command_executing` to be immediately before the I/O starts and after it completes. This technically shouldn't matter due to there not being any other instances of `await` between the existing locations of the assignments, but I agree that it's easier to read this way.